lms/security.py

```diff
@@ -99,10 +121,14 @@ def identity(self, request):
         ):
             permissions.append(Permissions.LTI_CONFIGURE_ASSIGNMENT)
 
-        return Identity(self._get_userid(lti_user), permissions)
+        return Identity(self._get_userid(lti_user), permissions, lti_user)
 
-    def permits(self, request, context, permission):
-        return _permits(self, request, context, permission)
+    def permits(self, request, _context, permission):
+        identity = self.identity(request)
+        if self._validation_error:
```

Comment:
```suggestion
        if self._validation_error is not None:
```

You might want to be a bit careful here because exceptions are sometimes falsey. IIRC this is true of Pyramid's built-in HTTP exception classes -- the ones for `{2,3}xx`'s are truthy but the ones for `{4,5}xx`'s are falsey -- and this has bitten us a couple of times in the past. I believe you're dealing with a custom validation error class here but still.